### ISSUE TITLE
Title Field Has Disappeared From the Create Blog Form

### ISSUE THREAD

**Issue by stephit** [backdrop/backdrop-issues#4015]
**Description of the bug**
I have a Drupal 7 website I just ported over to Backdrop - running the latest version, did all the db updates, etc. When trying to add a blog post, the title field is completely missing from the edit form. Titles show up on old blog nodes but there is also no way to edit them. Other content types do not seem to be affected, though I haven't checked every single one of them. I don't have any custom page templates set up for editing. Thanks for the help.

**Steps To Reproduce**
1. Go to 'create blog'

**Actual behavior**
- No title field is available. You can still create the blog node but no title is created. 
- If a description is added to the title field entry in the content types section, the description will show up but not the title field itself. 
- The title field is also not found when viewing the page source.

**Additional information**
- Backdrop CMS version: 1.13.3
- Web server and its version: Apache with PHP 7.1.30
- Browser(s) and their versions: Chrome, latest

**Comment by klonos** [comment]
@jenlampton have you faced anything like that before?

@stephit welcome to Backdrop, and thanks for taking the time to report this 👍 ...are there any errors logged in dblog? What happens if you enable all errors in `/admin/config/development/logging` and try to add/edit a blog node?

PS: is it possible to perhaps share a post-upgrade version of the D7 database (with sanitized user/names/passwords etc. of course)? It would be easier to figure out what's wrong if we could reproduce this.

**Comment by stephit** [comment]
OK, I was able to produce an error message by manually putting a title into the database. When viewing the node, it says:

Notice: Undefined offset: 0 in file_field_widget_form() (line 479 of [site]/core/modules/file/file.field.inc).

It looks like the disappearing title has something to do with old content types that previously had another module associated with them. I found 3 content types that have this issue - blog, forum, and products (from Ubercart). All of these content types had additional modules associated with its respective content type at one point or another. 

I had problems getting Ubercart to work with Backdrop, and I wasn't going to use it anyway. The forum had been shut off for ages (this was originally a Drupal 6 website). I'm not even sure when or what the blog module was in the past but I think it was integrated with core at some point? So none of these content types have those modules "on" now.

Also, it's possible there was some sort of upgrade path issue with older unused modules not getting updated properly first...but it's all 3 so there must be some sort of common denominator beyond one or even two of them skipping a database update. 

When I look at the node info in the backend it all looks consistent with everything else...not sure where else to look. 

PS I'm running my staging site on a shared host without Drush so it's probably not feasible for me to get a sanitized version of the site. 

Thank you!

**Comment by klonos** [comment]
> ... had additional modules associated ...

Care to elaborate? What modules, and associated in what way?

In general, this will be hard to figure out if nobody else had this happen to them, and if we have no way to reproduce 😞

**Comment by jenlampton** [comment]
No, I haven't seen that before, but I bet it has something to do with the node types being provided by modules in D7 (or D6). I bet they are either missing or incomplete in Backdrop. 

[Here is the code that builds the node form.](https://api.backdropcms.org/api/backdrop/core%21modules%21node%21node.pages.inc/function/node_form/1) The only part that does anything to the title element sets a weight, and only if the title already exists. That means the title field itself needs to be provided by the module that defined the node type. This is why a missing node type would create a missing title field. 

I bet that blog module defined a `blog_form()` that added the title field. Likewise for the other types. Without the node type, and/or these functions that add the title, we get no title.

**Comment by jenlampton** [comment]
I agree this is weird, but I don't think it's a bug. I think it's a data problem :)

For my upgrades I knew I would have types that were removed in backdrop. In my case, I changed the node type before I did the upgrade so that the upgrade would proceed as though the node types were custom node types.

[Here's an example of some code that converts the ad node type into a custom node type.](https://github.com/jenlampton/upgradenotes/blob/master/code/drupal6/acrupdate/acrupdate.install#L25) In this case, I also had to move the content of a field provided by ad module to a custom field (as provided by field module).

You might be able to get away with doing as little as [this one line of code](https://github.com/jenlampton/upgradenotes/blob/master/code/drupal6/acrupdate/acrupdate.install#L62) to fix your content types. You will need to adapt that to match whatever version of Drupal you run it on -- this code is for D6.
